src/states/elder/tests.rs

```diff
@@ -52,6 +54,30 @@ impl CandidateInfo {
     }
 }
 
+struct ClientInfo {
+    full_id: FullId,
+    addr: SocketAddr,
+}
+
+impl ClientInfo {
+    fn with_addr(addr: &str) -> Self {
```

Comment:
normally just `new` for the constructor?
`with_addr` indicating there is another constructor `without addr` ?

Reply:
I think `new` is idiomatic for a constructor that takes no argument. If you take an argument, especially a `&str` where the type isn't particularly meaningful, `with_addr` is a better name as it tells the user what argument they should pass with just the signature. The name of the variable is local and not part of the signature.

Reply:
I think I originally wanted to also nave `new` which would return a default or random address, but ended up not needing it. I'm fine with renaming it to `new`, but I think @pierrechevalier83 has a valid point.